backend/apps/github/models/user.py

```diff
@@ -1,46 +1,108 @@
+    GITHUB_NODE_ID_REQUIRED,
+    GITHUB_USER_DATA_REQUIRED,
+)
 from apps.github.models.common import GenericUserModel, NodeModel
 
 
 class User(NodeModel, GenericUserModel, TimestampedModel):
-    """User model."""
+    """Model representing a GitHub user or organization."""
 
     class Meta:
         db_table = "github_users"
         verbose_name_plural = "Users"
+        ordering = ["login"]
+        indexes = [
+            models.Index(fields=["login"]),
+            models.Index(fields=["type"]),
+        ]
 
-    bio = models.TextField(verbose_name="Bio", max_length=1000, default="")
-    is_hireable = models.BooleanField(verbose_name="Is hireable", default=False)
-    twitter_username = models.CharField(verbose_name="Twitter username", max_length=50, default="")
+    bio = models.TextField(
+        verbose_name="Bio",
+        max_length=1000,
+        blank=True,
+        default="",
+        help_text="User's GitHub biography",
+    )
+    type = models.CharField(
+        verbose_name="User type",
+        max_length=50,
+        default="User",
+        choices=[
+            ("User", "User"),
+            ("Organization", "Organization"),
+        ],
+        help_text="Distinguishes between User and Organization accounts",
+    )
+    public_repositories_count = models.PositiveIntegerField(
+        verbose_name="Public repositories count",
+        default=0,
+        help_text="Number of public repositories",
+    )
 
     def __str__(self):
-        """User human readable representation."""
+        """Return the string representation of the user."""
         return f"{self.name or self.login}"
 
+    def clean(self):
+        """Validate required fields before saving the user."""
+        super().clean()
+        if not self.login:
+            raise ValidationError(GITHUB_LOGIN_REQUIRED)
+        if not self.node_id:
+            raise ValidationError(GITHUB_NODE_ID_REQUIRED)
+
+    def save(self, *args, **kwargs):
+        """Save the user object after validation."""
+        self.full_clean()
+        super().save(*args, **kwargs)
+
     def from_github(self, gh_user):
-        """Update instance based on GitHub user data."""
+        """Populate user fields with data from GitHub."""
+        if not gh_user:
+            raise ValueError(GITHUB_USER_DATA_REQUIRED)
+
         super().from_github(gh_user)
 
         field_mapping = {
+            "avatar_url": "avatar_url",
+            "company": "company",
+            "email": "email",
+            "followers_count": "followers",
+            "following_count": "following",
+            "location": "location",
+            "login": "login",
+            "name": "name",
+            "created_at": "created_at",
+            "updated_at": "updated_at",
             "bio": "bio",
-            "is_hireable": "hireable",
-            "twitter_username": "twitter_username",
+            "type": "type",
+            "public_repositories_count": "public_repos",
```

Comment:
This is done in [GenericUserModel](https://github.com/OWASP/Nest/blob/4082718b8bf470557c38313595295724465254bc/backend/apps/github/models/common.py#L45)